contracts/implementations/nodes/Integration.sol

```diff
@@ -0,0 +1,287 @@
+//SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.13;
+
+import "../../interfaces/INFT.sol";
+import "../../libraries/NodesStorage.sol";
+import "../../libraries/nodes/IntegrationStorage.sol";
+
+import {DEFAULT_ADMIN_ROLE} from "../../shared/Roles.sol";
+import {AttributeInfoPair} from "../../shared/Types.sol";
+
+import "@solidstate/contracts/access/access_control/AccessControlInternal.sol";
+
+/// @title Integration
+/// @notice Contract that represents the Integration node
+contract Integration is AccessControlInternal {
+    event IntegrationIdProxySet(address indexed proxy);
+    event IntegrationAttributeAdded(string attribute);
+    event IntegrationAttributeSet(
+        uint256 tokenId,
```

Comment:
This, on the other hand, seems like an interested argument to index.

Reply:
Also fair